CodeGeneration/Tests/ValidateSyntaxNodes/Validation.swift

```diff
@@ -0,0 +1,136 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SyntaxSupport
+
+/// All nodes with base kind e.g. `ExprSyntax` should end with `ExprSyntax`.
+func validateBaseKindSuffix() -> [ValidationFailure] {
+  var failures: [ValidationFailure] = []
+  for node in SYNTAX_NODES where node.base != .syntaxCollection {
+    if !node.kind.syntaxType.description.hasSuffix(node.base.syntaxType.description) {
```

Comment:
I would also here add the if to the where? 
What do you think about that?

Reply:
Here I think I’d like to keep it separate because this is less about the nodes that we would like to check but about doing the actual validation.